packages/client/hooks/useGroupMatrix.ts

```diff
@@ -56,40 +56,38 @@ const useGroupMatrix = (
 
   const updateMatrix = () => {
     const matrixGroupsIds = groupMatrix.flatMap((row) => row.map(({id}) => id))
-    const newGroup = resultsGroups.find((group) => !matrixGroupsIds.includes(group.id))
-    const refGroupsIds = resultsGroups.map(({id}) => id)
-    const removedGroupId = matrixGroupsIds.find((id) => !refGroupsIds.includes(id))
-    if (newGroup && removedGroupId) {
-      // added to kanban group. Remove old Spotlight group and add kanban group in place
-      const newMatrix = groupMatrix.map((row) =>
-        row.map((group) => (group.id === removedGroupId ? newGroup : group))
-      )
-      setGroupMatrix(newMatrix)
-    } else if (newGroup) {
-      // ungrouping added a new group
-      const emptiestColumnIdx = getEmptiestColumnIdx()
-      const matrixColumns = groupMatrix.length - 1
-      const newMatrix =
-        emptiestColumnIdx > matrixColumns // add to empty column
-          ? [...groupMatrix, [newGroup]]
-          : groupMatrix.map((row, idx) => (idx === emptiestColumnIdx ? [...row, newGroup] : row))
-      setGroupMatrix(newMatrix)
-    } else if (removedGroupId) {
-      // group added to another Spotlight group. Remove the old group & remove column if empty
-      const newMatrix = groupMatrix
-        .map((row) => row.filter((group) => group.id !== removedGroupId))
-        .filter(({length}) => length)
+    const newGroups = resultsGroups.filter(({id}) => !matrixGroupsIds.includes(id))
+    const resultsGroupsIds = resultsGroups.map(({id}) => id)
+    const removedGroupsIds = matrixGroupsIds.filter((id) => !resultsGroupsIds.includes(id))
+    if (!removedGroupsIds.length && !newGroups.length) return
+    // grouping results or updating the search query can remove groups
+    const filteredMatrix = groupMatrix
+      .map((row) => row.filter(({id}) => !removedGroupsIds.includes(id)))
+      .filter(({length}) => length) // remove column if empty
+    if (!newGroups.length) {
```

Comment:
Previously, if a results group was grouped/ungrouped it would be added or removed from the matix. With the search query, many groups could be added or removed, so I've updated the logic.